qemu/tests/secure_img.py

```diff
@@ -0,0 +1,75 @@
+from virttest import error_context
+
+
+def run_cmd_in_guest(vm, cmd, test, timeout=60):
+    """
+    Run command in the guest
+
+    :params vm: vm object
+    :params cmd: a command needs to be ran
+    :params session: the vm's session
+    :params status: cmd status
+    :params output: cmd output
+    """
+    session = vm.wait_for_login(timeout=timeout)
+    status, output = session.cmd_status_output(cmd, timeout=timeout)
+    test.log.info("The command of '%s' output: %s", cmd, output)
+    if not status:
+        return output
+    else:
+        test.fail("cmd runs failed cmd: %s, output: %s", cmd, output)
+    session.close()
```

Comment:
`session.close()` is never executed because you either return or raise an exception (`test.fail`). Please, move it before the `if... else` block.

Reply:
Updated